src/data/real/ereal.lean

```diff
@@ -121,6 +125,38 @@ protected def mul : ereal → ereal → ereal
 
 instance : has_mul ereal := ⟨ereal.mul⟩
 
+/-- Induct on two ereals by performing case splits on the sign of one whenever the other is
+infinite. -/
+@[elab_as_eliminator]
+lemma induction₂ {P : ereal → ereal → Prop}
+  (t_t : P ⊤ ⊤)
+  (t_p : ∀ x : ℝ, 0 < x → P ⊤ x)
+  (t_z : P ⊤ 0)
+  (t_n : ∀ x : ℝ, x < 0 → P ⊤ x)
+  (t_b : P ⊤ ⊥)
+  (p_t : ∀ x : ℝ, 0 < x → P x ⊤)
+  (p_b : ∀ x : ℝ, 0 < x → P x ⊥)
+  (z_t : P 0 ⊤)
+  (h : ∀ x y : ℝ, P x y)
+  (z_b : P 0 ⊥)
+  (n_t : ∀ x : ℝ, x < 0 → P x ⊤)
+  (n_b : ∀ x : ℝ, x < 0 → P x ⊥)
+  (b_t : P ⊥ ⊤)
+  (b_p : ∀ x : ℝ, 0 < x → P ⊥ x)
+  (b_z : P ⊥ 0)
+  (b_n : ∀ x : ℝ, x < 0 → P ⊥ x)
+  (b_b : P ⊥ ⊥) :
```

Comment:
What do you think of slightly lengthening the case names for clarity? `bot`, `neg`, `zer`, `pos`, `top` looks like a good bet.

Reply:
Sure, I've done that; though it adds a fair number of lines.

Reply:
Unfortunate, but probably needed for future us' sake.